include/engine/datafacade/memory_datafacade_base.hpp

```diff
@@ -47,10 +44,10 @@ namespace engine
 namespace datafacade
 {
 
-class SharedDataFacade final : public BaseDataFacade
+class MemoryDataFacadeBase : public BaseDataFacade
```

Comment:
I find this name highly impractical. `memory` at least to me is anything you can store and read data from. So `Internal Memory`, usually referring to private RAM, `Shared Memory` to shared Ram or `External Memory` which collects all mounted devices.

Since All Datafacades (shared/internal) implement this interface, why do we need to put it between BaseDataFacade and the two users? Why don't we simply add the functionality in BaseDataFacade and avoid this additional layer.

Is there ever a use-case where we would have a BaseDataFacade without a MemoryDataFacade functionality?